### PR TITLE
added argument for Visual Studio version specification

### PR DESCRIPTION
**Pull request by MihaMarkic**
Defaults to latest, it lets you specify guys from VisualStudioVersion enum (minus VS prefix), so: 2012, 2013, 2015 and Latest.

#11

### CODE REVIEW

File: binder/Binder.cs [hunk 1]
```diff
@@ -64,45 +72,60 @@ static void ParseCommandLineArgs(string[] args)
 
         static GeneratorKind ConvertToGeneratorKind(string gen)
         {
-            switch(gen.ToLower())
+            switch (gen.ToLower())
```
Comment (tritao) [inline binder/Binder.cs]: Can you leave these formatting changes out of this commit? We can apply them afterwards in different commit as they are unrelated.

File: binder/Binder.cs [hunk 2]
```diff
@@ -64,45 +72,60 @@ static void ParseCommandLineArgs(string[] args)
...
-                return TargetPlatform.WatchOS;
-            case "tvos":
-                return TargetPlatform.TVOS;
+            if (string.Equals(version, "latest", StringComparison.InvariantCultureIgnoreCase))
+            {
+                return VisualStudioVersion.Latest;
+            }
+            VisualStudioVersion result;
+            if (Enum.TryParse("VS" + version, out result))
+            {
```
Comment (tritao) [inline binder/Binder.cs]: We do not use a block here in our code style since it's single statement.

File: binder/Binder.cs [hunk 3]
```diff
@@ -64,45 +72,60 @@ static void ParseCommandLineArgs(string[] args)
...
-            case "macos":
-                return TargetPlatform.MacOS;
-            case "ios":
-                return TargetPlatform.iOS;
-            case "watchos":
-                return TargetPlatform.WatchOS;
-            case "tvos":
-                return TargetPlatform.TVOS;
+            if (string.Equals(version, "latest", StringComparison.InvariantCultureIgnoreCase))
+            {
```
Comment (tritao) [inline binder/Binder.cs]: Same here.

File: binder/Compilation.cs [hunk 4]
```diff
@@ -398,7 +398,21 @@ void CompileMSVC(IEnumerable<string> files)
...
-            var vsSdk = vsSdks.LastOrDefault();
+            ToolchainVersion vsSdk;
+            if (Options.VsVersion == VisualStudioVersion.Latest)
+            {
+                vsSdk = vsSdks.LastOrDefault();
+            }
+            else
+            {
+                var exactVersion = vsSdks.Where(vs => vs.Version == (float)Options.VsVersion).Cast<ToolchainVersion?>().SingleOrDefault();
+                if (!exactVersion.HasValue)
```
Comment (tritao) [inline binder/Compilation.cs]: Same here with unnecessary block.

**Comment by tritao** [comment]
Looks mostly good, can you just apply the minor changes I requested and I will merge it, thanks for contributing.

**Comment by MihaMarkic** [comment]
Yeah, auto formatting. I think it should be good now.
BTW. I'd go with blocks anytime - otherwise you might get heartbleed in no time :)

**Comment by tritao** [comment]
Yeah once in a while some bug could fall through because of it, but I find the code way more readable without them and it also matches the Mono coding guidelines.
